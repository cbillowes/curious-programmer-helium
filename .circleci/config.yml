version: 2
jobs:
  build:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Chopping onions without the tears"
      - run: npm run build
  test:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making soup and drinking beers"
      - run: sudo npm install -g jest-cli
      - run: npm run test
  page_speed:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making sure it won't take years"
      - run: echo "* Deploy to Netlify"
      - run: echo "* Execute lighthouse page speed tests on build"
workflows:
  version: 2
  integration:
    jobs:
      - build
      - test:
          requires:
            - build
      - page_speed:
          requires:
            - test
