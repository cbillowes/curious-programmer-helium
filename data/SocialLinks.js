module.exports = {
  footer: [{
    name: "Github",
    href: "https://github.com/cbillowes",
    icon: "github"
  }, {
    name: "Twitter",
    href: "https://twitter.com/cbillowes",
    icon: "twitter"
  }, {
    name: "LinkedIn",
    href: "https://www.linkedin.com/in/cbouwer/",
    icon: "linkedin"
  }],
  contact: [{
    name: "email",
    href: "mailto:curiousprogrammer.io@gmail.com",
    icon: "email"
  }, {
    name: "RSS",
    href: "/rss.xml",
    icon: "rss"
  }, {
    name: "Twitter",
    href: "https://twitter.com/cbillowes",
    icon: "twitter"
  }, {
    name: "StackOverflow",
    href: "https://stackoverflow.com/users/849986/cbillowes",
    icon: "stack-overflow"
  }, {
    name: "Github",
    href: "https://github.com/cbillowes",
    icon: "github"
  }, {
    name: "LinkedIn",
    href: "https://www.linkedin.com/in/cbouwer/",
    icon: "linkedin"
  }]
}