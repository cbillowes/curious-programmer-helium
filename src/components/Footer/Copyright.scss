.footer-container {
    .footer {
        .copyright-footer {
            font-size: 0.8em;
        }
    }
}
