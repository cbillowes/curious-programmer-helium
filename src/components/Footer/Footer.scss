@import "../../layout/layout.scss";

.footer-container {
    background-color: $background;
    position: relative;

    .footer {
        font-family: $font-family-serif;
        margin: 0 auto;
        padding: 2em;
        font-weight: 100;
    }
}
