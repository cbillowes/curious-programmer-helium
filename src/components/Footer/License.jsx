import React, { Component } from "react"
import License from "../License/License"
import Link from "gatsby-link"
import "./License.scss"

class LicenseFooter extends Component {
  render() {
    return (
      <div className="license-footer">
        <License /> 
        {" "}
        <p>
          Also, there is a 
          {" "}
          <Link to="/license">License</Link>{" for the code "}
          and it's probably a good idea to read the
          {" "}
          <Link to="/privacy-policy">Privacy Policy</Link>
          .
        </p>
      </div>
    )
  }
}

export default LicenseFooter
