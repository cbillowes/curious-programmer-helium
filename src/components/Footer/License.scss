@import "../../layout/variables.scss";

.footer-container {
    .footer {
        .license-footer {
            font-size: 0.7em;
            font-style: italic;
            font-family: $font-family-serif;
            font-weight: 100;
            line-height: 1.75em;
        }

        p {
            margin: 0;
        }

        .baby {
            display: none;
        }

        // TODO: Get this 900px monstor into a variable,
        // better yet, the entire media query into a mixin
        @media screen and (max-width: 900px) {
            .baby {
                display: block;
            }

            .grand {
                display: none;
            }
        }
    }
}
