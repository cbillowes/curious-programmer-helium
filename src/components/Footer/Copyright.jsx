import React, { Component } from "react"
import Link from "gatsby-link"
import config from "../../../data/SiteConfig"
import "./Copyright.scss"

class License extends Component {
  render() {
    return (
      <div className="copyright-footer">
        Copyright &copy; { "" } 
        {new Date().getFullYear()} { "" }
        {config.siteTitle}. { "" }
        Made with Gatsby v2 and other { "" }
        <Link to="/credits">cool stuff</Link>.
      </div>
    )
  }
}

export default License
