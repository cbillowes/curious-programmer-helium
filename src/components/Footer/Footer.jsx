import React, { Component } from "react"
import License from "./License"
import Copyright from "./Copyright"
import SocialLinks from "./SocialLinks"
import config from "../../../data/SiteConfig"
import "./Footer.scss"

class Footer extends Component {
  render() {
    return (
      <div className="footer-container not-for-print">
        <footer className="footer">
          <SocialLinks />
          <Copyright />
          <License />
        </footer>
      </div>
    )
  }
}

export default Footer
