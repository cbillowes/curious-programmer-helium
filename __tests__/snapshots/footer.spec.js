
import React from "react"
import renderer from "react-test-renderer"
import Copyright from "../../src/components/Footer/Copyright"
import Footer from "../../src/components/Footer/Footer"
import License from "../../src/components/Footer/License"
import SocialLinks from "../../src/components/Footer/SocialLinks"
import config from "../../data/SiteConfig"

describe("footer components", () => {
  describe("snapshots", () => {
    it("component", () => {
      const tree = renderer.create(<Footer />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("copyright", () => {
      const tree = renderer.create(<Copyright />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("license", () => {
      const tree = renderer.create(<License />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("social links", () => {
      const tree = renderer.create(<SocialLinks />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})