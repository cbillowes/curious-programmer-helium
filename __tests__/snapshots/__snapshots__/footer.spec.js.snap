// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer components snapshots component 1`] = `
<div
  className="footer-container not-for-print"
>
  <footer
    className="footer"
  >
    <ul
      className="social-links-footer"
    >
      <li
        className="link"
      >
        <a
          className="github"
          href="https://github.com/cbillowes"
          rel="nofollow noopener noreferrer "
          target="_blank"
        />
      </li>
      <li
        className="link"
      >
        <a
          className="twitter"
          href="https://twitter.com/cbillowes"
          rel="nofollow noopener noreferrer "
          target="_blank"
        />
      </li>
      <li
        className="link"
      >
        <a
          className="linkedin"
          href="https://www.linkedin.com/in/cbouwer/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        />
      </li>
    </ul>
    <div
      className="copyright-footer"
    >
      Copyright © 
      
      2019
       
      
      Curious Programmer
      . 
      
      Made with Gatsby v2 and other 
      
      <a
        href="/credits"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        cool stuff
      </a>
      .
    </div>
    <div
      className="license-footer"
    >
      <span
        className="license"
      >
        <a
          className="license-icon"
          href="http://creativecommons.org/licenses/by-sa/4.0/"
          rel="nofollow noopener noreferrer license"
          target="_blank"
        >
          <img
            alt="Creative Commons License"
            src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
            title="Creative Commons Attribution-ShareAlike 4.0 International License"
          />
        </a>
        <p>
          <span
            property="dct:title"
            xmlns:dct="http://purl.org/dc/terms/"
          >
             
            Curious Programmer
             content
             
          </span>
           
          by
           
          <a
            className=""
            href="https://curiousprogrammer.io"
            rel="nofollow noopener noreferrer cc:attributionURL"
            target="_blank"
          >
            Clarice Bouwer
          </a>
           
          is licensed under a
           
          <a
            className=""
            href="http://creativecommons.org/licenses/by-sa/4.0/"
            rel="nofollow noopener noreferrer license"
            target="_blank"
          >
            Creative Commons Attribution-ShareAlike 4.0 International License
          </a>
          .
        </p>
      </span>
       
      <p>
        Also, there is a
         
        <a
          href="/license"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          License
        </a>
         for the code 
        and it's probably a good idea to read the
         
        <a
          href="/privacy-policy"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Privacy Policy
        </a>
        .
      </p>
    </div>
  </footer>
</div>
`;

exports[`footer components snapshots copyright 1`] = `
<div
  className="copyright-footer"
>
  Copyright © 
  
  2019
   
  
  Curious Programmer
  . 
  
  Made with Gatsby v2 and other 
  
  <a
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    cool stuff
  </a>
  .
</div>
`;

exports[`footer components snapshots license 1`] = `
<div
  className="license-footer"
>
  <span
    className="license"
  >
    <a
      className="license-icon"
      href="http://creativecommons.org/licenses/by-sa/4.0/"
      rel="nofollow noopener noreferrer license"
      target="_blank"
    >
      <img
        alt="Creative Commons License"
        src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
        title="Creative Commons Attribution-ShareAlike 4.0 International License"
      />
    </a>
    <p>
      <span
        property="dct:title"
        xmlns:dct="http://purl.org/dc/terms/"
      >
         
        Curious Programmer
         content
         
      </span>
       
      by
       
      <a
        className=""
        href="https://curiousprogrammer.io"
        rel="nofollow noopener noreferrer cc:attributionURL"
        target="_blank"
      >
        Clarice Bouwer
      </a>
       
      is licensed under a
       
      <a
        className=""
        href="http://creativecommons.org/licenses/by-sa/4.0/"
        rel="nofollow noopener noreferrer license"
        target="_blank"
      >
        Creative Commons Attribution-ShareAlike 4.0 International License
      </a>
      .
    </p>
  </span>
   
  <p>
    Also, there is a
     
    <a
      href="/license"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      License
    </a>
     for the code 
    and it's probably a good idea to read the
     
    <a
      href="/privacy-policy"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Privacy Policy
    </a>
    .
  </p>
</div>
`;

exports[`footer components snapshots social links 1`] = `
<ul
  className="social-links-footer"
>
  <li
    className="link"
  >
    <a
      className="github"
      href="https://github.com/cbillowes"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
  <li
    className="link"
  >
    <a
      className="twitter"
      href="https://twitter.com/cbillowes"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
  <li
    className="link"
  >
    <a
      className="linkedin"
      href="https://www.linkedin.com/in/cbouwer/"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
</ul>
`;
